### PR TITLE
Ref #17769 - Use ServerRequest Object to access POST-vars

### PR DESCRIPTION
**Pull request by MrMadClown**
### Description

As described in #17769

### CODE REVIEW

File: libraries/classes/Controllers/Table/TrackingController.php [hunk 1]
```diff
@@ -65,13 +65,14 @@ public function __invoke(ServerRequest $request): void
...
+
         if (
             Tracker::isActive()
             && Tracker::isTracked($GLOBALS['db'], $GLOBALS['table'])
-            && ! (isset($_POST['toggle_activation'])
-                && $_POST['toggle_activation'] === 'deactivate_now')
-            && ! (isset($_POST['report_export'])
-                && $_POST['export_type'] === 'sqldumpfile')
+            && ! ($toggleActivation === 'deactivate_now')
+            && ! ($reportExport === 'sqldumpfile')
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Table/TrackingController.php:75]:
```suggestion
            && $reportExport !== 'sqldumpfile'
```

File: libraries/classes/Controllers/Table/TrackingController.php [hunk 2]
```diff
@@ -65,13 +65,14 @@ public function __invoke(ServerRequest $request): void
...
+        $reportExport = $request->getParsedBodyParam('report_export');
+
         if (
             Tracker::isActive()
             && Tracker::isTracked($GLOBALS['db'], $GLOBALS['table'])
-            && ! (isset($_POST['toggle_activation'])
-                && $_POST['toggle_activation'] === 'deactivate_now')
-            && ! (isset($_POST['report_export'])
-                && $_POST['export_type'] === 'sqldumpfile')
+            && ! ($toggleActivation === 'deactivate_now')
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Table/TrackingController.php:74]:
```suggestion
            && $toggleActivation !== 'deactivate_now'
```

File: libraries/classes/Controllers/Table/TrackingController.php [hunk 3]
```diff
@@ -138,15 +134,17 @@ public function __invoke(ServerRequest $request): void
...
 
         $actionMessage = '';
-        if (isset($_POST['submit_mult'])) {
-            if (! empty($_POST['selected_versions'])) {
-                if ($_POST['submit_mult'] === 'delete_version') {
-                    foreach ($_POST['selected_versions'] as $version) {
+        $submitMult = $request->getParsedBodyParam('submit_mult');
+        $selectedVersions = $request->getParsedBodyParam('selected_versions');
+        if ($submitMult !== null) {
+            if (is_array($selectedVersions) && count($selectedVersions) !== 0) {
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Table/TrackingController.php:145]:
```suggestion
            if (is_array($selectedVersions) && $selectedVersions !== []) {
```

File: libraries/classes/Controllers/Table/TrackingController.php [hunk 4]
```diff
@@ -162,79 +160,80 @@ public function __invoke(ServerRequest $request): void
...
         $deactivateTracking = '';
-        if (isset($_POST['toggle_activation']) && $_POST['toggle_activation'] === 'deactivate_now') {
+        $activateTracking = '';
+
+        if ($toggleActivation === 'deactivate_now') {
             $deactivateTracking = $this->tracking->changeTracking(
                 $GLOBALS['db'],
                 $GLOBALS['table'],
-                $_POST['version'],
+                $request->getParsedBodyParam('version'),
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Table/TrackingController.php:187]: Maybe create a new variable for this instead of getting the param many times?

**Review by kamil-tekiela — CHANGES_REQUESTED** [review]
Just few small fixes.

File: libraries/classes/Controllers/Server/ReplicationController.php [hunk 5]
```diff
@@ -91,23 +92,23 @@ public function __invoke(ServerRequest $request): void
...
         if ($primaryInfo['status']) {
             $primaryReplicationHtml = $this->replicationGui->getHtmlForPrimaryReplication(
-                $_POST['primary_connection'] ?? null,
+                $request->getParsedBodyParam('primary_connection'),
                 $params['repl_clear_scr'],
-                $_POST['primary_add_user'] ?? null,
-                $_POST['username'] ?? null,
-                $_POST['hostname'] ?? null
+                $request->getParsedBodyParam('primary_add_user'),
+                 $request->getParsedBodyParam('username'),
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Server/ReplicationController.php:98]:
```suggestion
                $request->getParsedBodyParam('username'),
```

File: libraries/classes/Controllers/Server/ReplicationController.php [hunk 6]
```diff
@@ -91,23 +92,23 @@ public function __invoke(ServerRequest $request): void
...
+                 $request->getParsedBodyParam('hostname')
```
Comment (kamil-tekiela) [inline libraries/classes/Controllers/Server/ReplicationController.php:99]:
```suggestion
                $request->getParsedBodyParam('hostname')
```

File: libraries/classes/Controllers/Server/ReplicationController.php [hunk 7]
```diff
@@ -91,23 +92,23 @@ public function __invoke(ServerRequest $request): void
...
             );
         }
 
         if (isset($params['primary_configure'])) {
             $primaryConfigurationHtml = $this->replicationGui->getHtmlForPrimaryConfiguration();
         } else {
-            if (! isset($params['repl_clear_scr'])) {
+            if (!isset($params['repl_clear_scr'])) {
```
[the remark this answers is not in the document]
Comment (MrMadClown) [reply libraries/classes/Controllers/Server/ReplicationController.php:106]: This was probably me hitting auto-reformat in the IDE out of habit, same with the alignments above

**Review by kamil-tekiela — APPROVED** [review]
LGTM but please fix the remaining coding style issues https://github.com/phpmyadmin/phpmyadmin/actions/runs/3294855920/jobs/5432806072

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**